Runtime/System/ClientSimPlayerManager.cs

```diff
@@ -322,6 +322,12 @@ public static void TeleportTo(VRCPlayerApi player, Vector3 position, Quaternion
         {
             TeleportToOrientationLerp(player, position, rotation, VRC_SceneDescriptor.SpawnOrientation.Default, false);
         }
+        
+        
+        public static void Respawn(VRCPlayerApi playerApi)
+        {
+            playerApi.GetPlayerController().Respawn();
```

Comment:
I believe this will crash if the player isn't local as GetPlayerController() will return null.
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/Utilities/ClientSimExtensions.cs#L10-L23

While this solution will work and is probably the minimum amount of new coded added, I wonder about directly exposing the Respawn method. Convention in ClientSim seems a little mixed between direct exposing and using the EventDispatcher to send and handle different events. It is up to you on which version you should use. 

If you go the event route, I recommend moving this to ClientSimMain and creating a non-static method that dispatches the event. Here are some references to the Respawn event used in the menu as an example:

Respawn from menu event:
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/Events/EventTypes/ClientSimMenuEvents.cs#L8

Menu sending the respawn event:
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/System/ClientSimMenu.cs#L415-L420

Player controller's handling of the menu respawn event:
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/Player/ClientSimPlayerController.cs#L154-L157

The player controller subscribes to events here:
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/Player/ClientSimPlayerController.cs#L99-L122

Player specific events are located in this file and a generic player respawn should be put here:
https://github.com/vrchat-community/ClientSim/blob/main/Runtime/Events/EventTypes/ClientSimOnPlayerEvents.cs

Reply:
Now checking for isLocal first, and logging a warning when trying to Respawn remote players.
Looking at the way you've implemented similar methods, I think this makes sense and matches the other code. If you have ideas / logic about when we should use one or the other, let me know so we can build the rules around it.

Reply:
This implementation is fine and matches other areas. If it was something else that didn't have an exposed way to get the object, I feel that case should instead use the EventDispatcher to prevent introducing new dependencies.